Google.Ads.GoogleAds.Core/src/Logging/LoggingHandler.cs

```diff
@@ -82,11 +82,17 @@ internal LoggingHandler()
             where TRequest : class
             where TResponse : class
         {
-            // Generating log entry is expensive, so let's do that only if the log source
-            // has been configured to do so.
-
             RpcException exception = null;
```

Comment:
Should we put this behind a check for performance concerns?

var genSummaryLogs = TraceUtilities.ShouldGenerateSummaryRequestLogs();
var genDetailLogs = TraceUtilities.ShouldGenerateDetailedRequestLogs();
var genAnyLogs = genSummaryLogs || genDetailLogs;

if (!genAnyLogs) {
  return;
}

Reply:
I don't think we are incurring a huge performance hit here, but i don't see any disadvantages in doing that check.

Added it.